Bottom: ab1609d226247298a7c944631319315efbe8561c
Top:    41fafd48989de6bfdba6f02016d05ad6d47887ef
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-06-15 16:12:44 +0800

Refresh of tun-introduce-tx-array-mode

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index d91c5f6..b22e475 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1533,7 +1533,7 @@ static struct sk_buff *tun_ring_recv(struct tun_file *tfile, int noblock,
 	add_wait_queue(&tfile->wq.wait, &wait);
 	current->state = TASK_INTERRUPTIBLE;
 
-	do {
+	while (1) {
 		skb = skb_array_consume(&tfile->tx_array);
 		if (skb)
 			break;
@@ -1547,7 +1547,7 @@ static struct sk_buff *tun_ring_recv(struct tun_file *tfile, int noblock,
 		}
 
 		schedule();
-	} while(skb);
+	};
 
 	current->state = TASK_RUNNING;
 	remove_wait_queue(&tfile->wq.wait, &wait);
